### PR TITLE
[#12] [Backend] Implement Home service

### LINKED ISSUE #12: [Backend] As a user, I can see my surveys
## Why

- Once logged in, users can navigate through a list of surveys that they can take. The mobile app will show the basic info of survey as they scroll through each survey.
 
## Acceptance Criteria
```
Call GET /api/v1/surveys endpoint with the following query parameters:
page[number]: 1
page[size]: 10
```
## Resources
[Postman - Nimble Survey API](https://documenter.getpostman.com/view/11835486/T1LJjTqj#2dd12ea2-3ecb-46d5-a48b-64256346e4e1)

### PR DESCRIPTION
**Pull request by nkhanh44**
#12 

## What happened

- Implemented API get survey list for Home Screen.
 
## Insight
- Implemented `AuthenInterceptor` to automatically add `accessToken` into the header.
- Created `AuthenNetworkAPI` that conforms `NetworkAPIProtocol` for requests that needs authenticated
 
## Proof Of Work

OnSuccess:
Screen Shot 2022-08-17 at 16 14 50

### CODE REVIEW

File: Survey.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -407,6 +419,15 @@
 			path = Configurations;
 			sourceTree = "<group>";
 		};
+		1C044B9B28ACD82A001FF098 /* Recovered References */ = {
```
Comment (blyscuit) [inline Survey.xcodeproj/project.pbxproj]: Please check and remove this ref.

File: Survey.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1560,10 +1590,13 @@
 				1C94DA1228996C8100ED0984 /* SError.swift in Sources */,
 				1CB18675288E534400372D50 /* ActivityTracker.swift in Sources */,
 				1CD4E37828A0CF6200EA1E8B /* StoreTokenUseCase.swift in Sources */,
+				1C044B9D28ACD9A5001FF098 /* HomeUseCase.swift in Sources */,
 				1C94DA1E289BBC9B00ED0984 /* ErrorTextModifier.swift in Sources */,
 				1CEE602828A4ABE100748C63 /* BodyTextModifier.swift in Sources */,
 				1C94DA1A289BA04800ED0984 /* ErrorResponse.swift in Sources */,
 				1CEE603028A4B77D00748C63 /* HeaderHomeView.swift in Sources */,
+				1C044B9F28ACDF58001FF098 /* AuthenInterceptor.swift in Sources */,
```
Comment (blyscuit) [inline Survey.xcodeproj/project.pbxproj]: I suggest using word in the dictionary for naming `AuthenticationInterceptor` `Authen` has no meaning and could create ambiguity.

**Review by blyscuit — APPROVED** [review]
Looks good. Please create issue for unit test if not done so. 🙏

File: Survey/Sources/Data/NetworkAPI/Core/AuthenNetworkAPI.swift [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+//  Survey
+//
+//  Created by Khanh on 17/08/2022.
+//  Copyright © 2022 Nimble. All rights reserved.
+//
+
+import Alamofire
+import Combine
+
+final class AuthenNetworkAPI: NetworkAPIProtocol {
```
Comment (ducbm051291) [inline Survey/Sources/Data/NetworkAPI/Core/AuthenNetworkAPI.swift]: I suggest `AuthenticationNetworkAPI`. What do you think? 
```suggestion
final class AuthenticationNetworkAPI: NetworkAPIProtocol {
```

**Review by ducbm051291 — COMMENTED** [review]
The rest LGTM

**Review by blyscuit — APPROVED** [review]

**Review by ducbm051291 — APPROVED** [review]

**Pull request merged**